baselines/hfedxgboost/hfedxgboost/dataset_preparation.py

```diff
@@ -0,0 +1,263 @@
+                with bz2.BZ2File(abs_filepath) as fr, open(
+                    abs_filepath[:-4], "wb"
+                ) as fw:
+                    shutil.copyfileobj(fr, fw)
+        # training then test ✅
+        return [
+            os.path.join(DATASET_PATH, "ijcnn1.t"),
+            os.path.join(DATASET_PATH, "ijcnn1.tr"),
+        ]
+
+    if dataset_name == "space_ga":
+        DATASET_PATH = os.path.join(ALL_DATASETS_PATH, "space_ga")
+        if not os.path.exists(DATASET_PATH):
+            os.makedirs(DATASET_PATH)
+            urllib.request.urlretrieve(
+                "https://www.csie.ntu.edu.tw/~cjlin/libsvmtools/datasets/regression/space_ga",
+                f"{os.path.join(DATASET_PATH, 'space_ga')}",
+            )
+            urllib.request.urlretrieve(
+                "https://www.csie.ntu.edu.tw/~cjlin/libsvmtools/datasets/regression/space_ga_scale",
+                f"{os.path.join(DATASET_PATH, 'space_ga_scale')}",
+            )
+        return [os.path.join(DATASET_PATH, "space_ga_scale")]
+    if dataset_name == "abalone":
+        DATASET_PATH = os.path.join(ALL_DATASETS_PATH, "abalone")
+        if not os.path.exists(DATASET_PATH):
+            os.makedirs(DATASET_PATH)
+            urllib.request.urlretrieve(
+                "https://www.csie.ntu.edu.tw/~cjlin/libsvmtools/datasets/regression/abalone",
+                f"{os.path.join(DATASET_PATH, 'abalone')}",
+            )
+            urllib.request.urlretrieve(
+                "https://www.csie.ntu.edu.tw/~cjlin/libsvmtools/datasets/regression/abalone_scale",
+                f"{os.path.join(DATASET_PATH, 'abalone_scale')}",
+            )
+        return [os.path.join(DATASET_PATH, "abalone_scale")]
+    if dataset_name == "cpusmall":
+        DATASET_PATH = os.path.join(ALL_DATASETS_PATH, "cpusmall")
+        if not os.path.exists(DATASET_PATH):
+            os.makedirs(DATASET_PATH)
+            urllib.request.urlretrieve(
+                "https://www.csie.ntu.edu.tw/~cjlin/libsvmtools/datasets/regression/cpusmall",
+                f"{os.path.join(DATASET_PATH, 'cpusmall')}",
+            )
+            urllib.request.urlretrieve(
+                "https://www.csie.ntu.edu.tw/~cjlin/libsvmtools/datasets/regression/cpusmall_scale",
+                f"{os.path.join(DATASET_PATH, 'cpusmall_scale')}",
+            )
+        return [os.path.join(DATASET_PATH, "cpusmall_scale")]
+    if dataset_name == "YearPredictionMSD":
+        DATASET_PATH = os.path.join(ALL_DATASETS_PATH, "YearPredictionMSD")
+        if not os.path.exists(DATASET_PATH):
+            print(
+                "long dowenload coming, it will be better if you dowenloaded",
+                "those 2 files manually with a faster dowenload manger program or",
+                "something and just place them in the rigte folder then get",
+                "the for loop out of the if condition to alter thier format",
+            )
```

Comment:
there are a couple of typos. Also probably worth saying that the download is ~615MB (in some context, this might be fine -- not so large)